File: 백준/Silver/14469. 소가 길을 건너간 이유 3/소가 길을 건너간 이유 3.py
import sys

N = int(sys.stdin.readline())
arr = [list(map(int, sys.stdin.readline().split())) for _ in range(N)]

arr.sort()

sum = arr[0][0] + arr[0][1]

for i in range(N):
    if i == 0:
        continue
    if arr[i][0] <= sum:
        sum += arr[i][1]
    else:
        sum = arr[i][0] + arr[i][1]
        
print(sum)

File: 백준/Silver/14469. 소가 길을 건너간 이유 3/소가 길을 건너간 이유 3.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {


    public static void main(String[] args) throws IOException {

        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(bf.readLine());
        int[][] arr = new int[N][2];

        for (int i = 0; i < N; i++) {
            StringTokenizer st = new StringTokenizer(bf.readLine());
            arr[i][0] = Integer.parseInt(st.nextToken());
            arr[i][1] = Integer.parseInt(st.nextToken());
        }

        Arrays.sort(arr, (o1, o2) -> {
            return o1[0] - o2[0];
        });

        int time = 0;

        time = time + arr[0][0];

        for (int i = 0; i < N; i++) {
            time = time + arr[i][1];

            if (i < N - 1 && arr[i + 1][0] > time) {
                time = time + (arr[i + 1][0] - time);
            }
        }

        System.out.println(time);


    }
}
